### PR TITLE
Update Kubernetes Model to v1.21.0

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
+ Update Kubernetes Model Schema as per Kubernetes v1.21.0

https://kubernetes.io/blog/2021/04/08/kubernetes-1-21-release-announcement/

##### DSL Changes:
- `client.batch().jobs()` deprecated
- `client.batch().cronjobs()` deprecated
- `client.policy().podSecurityPolicies()` deprecated
- `client.policy().podDisruptionBudget()` deprecated

##### Model Changes:
- Classes in `io.fabric8.kubernetes.api.model.batch` have been moved to `io.fabric8.kubernetes.api.model.batch.v1` and `io.fabric8.kubernetes.api.model.batch.v1beta1`
- Classes in `io.fabric8.kubernetes.api.model.policy` have been moved to `io.fabric8.kubernetes.api.model.policy.v1` and `io.fabric8.kubernetes.api.model.policy.v1beta1`
- Added `io.fabric8.kubernetes.api.model.discovery.v1` EndpointSlice

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java [hunk 1]
```diff
@@ -38,12 +31,12 @@ public BatchAPIGroupClient(OkHttpClient httpClient, final Config config) {
...
   @Override
-  public MixedOperation<Job, JobList, ScalableResource<Job>> jobs() {
-    return new JobOperationsImpl(httpClient, getConfiguration());
+  public V1BatchAPIGroupDSL v1() {
+    return adapt(V1BatchAPIGroupClient.class);
   }
 
   @Override
-  public MixedOperation<CronJob, CronJobList, Resource<CronJob>> cronjobs() {
-    return new CronJobOperationsImpl(httpClient, getConfiguration());
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java:47]: Could we keep this as `@Deprecated+@deprecated` and point to the old `V1beta1BatchAPIGroupClient` cronjobs and jobs?

This way changes won't be that breaking.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/PolicyAPIGroupClient.java [hunk 2]
```diff
@@ -33,12 +27,12 @@ public PolicyAPIGroupClient(OkHttpClient httpClient, final Config config) {
...
+  public V1PolicyAPIGroupDSL v1() {
+    return adapt(V1PolicyAPIGroupClient.class);
   }
 
   @Override
-  public MixedOperation<PodDisruptionBudget, PodDisruptionBudgetList, Resource<PodDisruptionBudget>> podDisruptionBudget() {
-    return new PodDisruptionBudgetOperationsImpl(httpClient, getConfiguration());
+  public V1beta1PolicyAPIGroupDSL v1beta1() {
+    return adapt(V1beta1PolicyAPIGroupClient.class);
   }
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/PolicyAPIGroupClient.java:37]: Same as https://github.com/fabric8io/kubernetes-client/pull/2979/files#r612202961

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java (hunk 1 above)
Comment (manusa) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java]: I don't know how you reverted these changes, but you reverted the examples too (not sure if that was on purpose).

I do think that examples should use the new non-deprecated APIs

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**